DriveKitTripAnalysisUI/TripSharing/View/TripSharingViewController.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  TripSharingViewController.swift
+//  DriveKitTripAnalysisUI
+//
+//  Created by Amine Gahbiche on 02/12/2024.
+//  Copyright © 2024 DriveQuant. All rights reserved.
+//
+
+import UIKit
+import DriveKitCommonUI
+
+class TripSharingViewController: DKUIViewController {
+    
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var activateShareButton: UIButton!
+    @IBOutlet weak var oneDayButton: UIButton!
+    @IBOutlet weak var oneWeekButton: UIButton!
+    @IBOutlet weak var oneMonthButton: UIButton!
+    @IBOutlet weak var cancelButton: UIButton!
+    @IBOutlet weak var shareLinkButton: UIButton!
+    @IBOutlet weak var revokeLinkButton: UIButton!
+
+    @IBOutlet weak var notActiveStackView: UIStackView!
+    @IBOutlet weak var selectPeriodStackView: UIStackView!
+    @IBOutlet weak var activeStackView: UIStackView!
+
+    private let viewModel: TripSharingViewModel
+    
+    init() {
+        self.viewModel = TripSharingViewModel()
+        super.init(nibName: String(describing: TripSharingViewController.self), bundle: Bundle.tripAnalysisUIBundle)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setup()
+        showLoader()
+        self.viewModel.updateStatus {
+            DispatchQueue.dispatchOnMainThread {
+                self.hideLoader()
+                self.updateView()
+            }
+        }
+    }
+
+    func setup() {
+        self.title = "dk_location_sharing_title".dkTripAnalysisLocalized()
+        self.view.backgroundColor = .white
+        
+        self.activateShareButton.configure(title: "dk_location_sharing_inactive_enable".dkTripAnalysisLocalized(), style: .full)
+        self.oneDayButton.configure(title: "dk_location_sharing_select_day".dkTripAnalysisLocalized(), style: .full)
+        self.oneWeekButton.configure(title: "dk_location_sharing_select_week".dkTripAnalysisLocalized(), style: .full)
+        self.oneMonthButton.configure(title: "dk_location_sharing_select_month".dkTripAnalysisLocalized(), style: .full)
+        self.cancelButton.setAttributedTitle(
+            "dk_location_sharing_select_cancel"
+                .dkTripAnalysisLocalized()
+                .dkAttributedString()
+                .font(dkFont: .primary, style: .button)
+                .color(.criticalColor)
+                .build(),
+            for: .normal
+        )
+        self.shareLinkButton.configure(title: "dk_location_sharing_active_share_link".dkTripAnalysisLocalized().uppercased(), style: .full)
```

Comment:
```suggestion
        self.shareLinkButton.configure(title: "dk_location_sharing_active_share_link".dkTripAnalysisLocalized(), style: .full)
```
(normalement ce n'est pas utile car effectué par le style `.full`)

Reply:
merci, corrigé sur b3bfd2cffcf8589bc4ba87d6a61254df20618549